src/components/DynamicDimension/ItemSelector.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Transfer } from '@dhis2/ui-core'
+import i18n from '@dhis2/d2-i18n'
+
+import styles from '../styles/DimensionSelector.style'
+import { TransferOption } from '../TransferOption'
+import GenericIcon from '../../assets/DimensionItemIcons/GenericIcon'
+import {
+    TRANSFER_HEIGHT,
+    TRANSFER_OPTIONS_WIDTH,
+    TRANSFER_SELECTED_WIDTH,
+} from '../../modules/dimensionSelectorHelper'
+
+class ItemSelector extends Component {
```

Comment:
Why not making also this a functional component?

Reply:
fair point, fixed!